permchain/langgraph/__init__.py

```diff
@@ -0,0 +1,110 @@
+from langchain_core.runnables import Runnable, RunnableMap
+from typing import Callable, Union, Optional
+from permchain import Channel, Pregel
+
+class Actor:
+
+    def __init__(self, name: str, runnable: Runnable):
```

Comment:
I wonder if there's an alternative setup here where actor looks like:

```
class Actor:
  def __init__(self, key: str, runnable: Runnable, input_keys: list[str], entry: bool):
    ....
```

That would mean you could skip all the `connect` calls and just set up your inputs inline. For decisionPoints you could have something similar, but maybe each branch has its own key (e.g. `{decisionPoint.key}.{conditionKey}`).

You'd be able to support circular dependencies too since you are doing this by key instead of by value.